src/XIVLauncher.Common.Unix/UnixDalamudRunner.cs

```diff
@@ -74,43 +74,57 @@ public UnixDalamudRunner(CompatibilityTools compatibility, DirectoryInfo dotnetR
         launchArguments.Add(gameArgs);
 
         var dalamudProcess = compatibility.RunInPrefix(string.Join(" ", launchArguments), environment: environment, redirectOutput: true, writeLog: true);
-        var output = dalamudProcess.StandardOutput.ReadLine();
+        var output = dalamudProcess.StandardOutput.ReadLine() ?? "";
 
-        if (output == null)
-            throw new DalamudRunnerException("An internal Dalamud error has occured");
+        Console.WriteLine("[DALAMUD] " + output);
+
+        // Skip "ERROR: Could Not Get Primary Adapter Handle" and proceed to next line
+        if (output.Equals("ERROR: Could Not Get Primary Adapter Handle"))
+        {
+            output = dalamudProcess.StandardOutput.ReadLine() ?? "";
+        }
 
-        Console.WriteLine(output);
+        if (string.IsNullOrEmpty(output) || output.Contains("ERROR"))
+            throw new DalamudRunnerException("An internal Dalamud error has occured");
 
         new Thread(() =>
         {
             while (!dalamudProcess.StandardOutput.EndOfStream)
             {
                 var output = dalamudProcess.StandardOutput.ReadLine();
                 if (output != null)
-                    Console.WriteLine(output);
+                    Console.WriteLine("[DALAMUD] " + output);
             }
 
         }).Start();
 
+        int unixPid = 0;
+        int winePid = 0;
         try
         {
             var dalamudConsoleOutput = JsonConvert.DeserializeObject<DalamudConsoleOutput>(output);
-            var unixPid = compatibility.GetUnixProcessId(dalamudConsoleOutput.Pid);
```

Comment:
All of `UnixDalamudRunner` can be left as is with the adjustments to `GetUnixProcessId` from before

Reply:
If `GetUnixProcessId` is properly implemented all of this can be really left as as (save the JSON reading adjustment  and spelling mistakes) to keep it simple